src/fable/Fable.Core/npm/fable-core.ts

```diff
@@ -3567,6 +3581,8 @@ const AsyncImpl = {
     return (ctx: IAsyncContext<T>) => {
       if (ctx.cancelToken.isCancelled)
         ctx.onCancel("cancelled");
+      else if (ctx.trampoline.incrementAndCheck()) 
+        ctx.trampoline.hijack(() => f(ctx))
```

Comment:
Maybe we should wrap the callback also in a try/catch as [it was done in FunScript](https://github.com/ZachBray/FunScript/blob/master/src/main/FunScript/Core/Async.fs#L38-L39).